### DISCUSSION TITLE
How to Imporvement msvc build?

### DISCUSSION THREAD

**Discussion by cathaysia** [microsoft/vcpkg#19399] [Q&A]
I have two question:

- Can I use ninja or multithread build?
- Can I use clang or specify utf-8 as the source encoding?

cl.exe build too slow, especially building getext[core,tools], about two hours on my machine. Have any way to speed it up? Use ninja or multithread build?

In addition, can I use clang build? cl.exe use gbk as the default encoding for source on my machine, but it maybe cause a error when source file contains any emojis or Chinese character.

**Comment by JackBoosY** [comment]
Unfortunately, `vcpkg_install_make` can only use msbuild in windows at present.
If we want to use ninja, we must add support for cmake.
